bundles/org.openhab.core/src/main/java/org/openhab/core/service/StartLevelService.java

```diff
@@ -249,10 +245,35 @@ private void setStartLevel(int level) {
             readyService.markReady(marker);
         }
         openHABStartLevel = level;
+        scheduler.submit(() -> {
+            for (StartLevelTracker slTracker : slTrackers) {
+                try {
+                    slTracker.onStartLevelChanged(level);
+                } catch (Exception e) {
+                    logger.warn("Failed to notify start level tracker: {}: {}", e.getClass(), e.getMessage());
+                }
+            }
+            logger.debug("Notified start level {}", level);
+        });
+
         scheduler.submit(() -> {
             StartlevelEvent startlevelEvent = SystemEventFactory.createStartlevelEvent(level);
             eventPublisher.post(startlevelEvent);
             logger.debug("Reached start level {}", level);
         });
     }
+
+    @Reference(policy = ReferencePolicy.DYNAMIC, cardinality = ReferenceCardinality.MULTIPLE, unbind = "unregisterTracker")
+    public void registerTracker(StartLevelTracker tracker) {
+        slTrackers.add(tracker);
+        tracker.onStartLevelChanged(openHABStartLevel);
+    }
+
+    public void unregisterTracker(StartLevelTracker tracker) {
+        slTrackers.remove(tracker);
+    }
+
+    public interface StartLevelTracker {
```

Comment:
I deliberately didn't introduce a dedicated tracker here as it somewhat duplicates the already existing `ReadyTracker` - this service can be used to track start level changes, see [this example](https://github.com/openhab/openhab-core/blob/127724c0e31ad8abf0855d87fbf96204267a39be/bundles/org.openhab.core.model.rule/src/org/openhab/core/model/rule/jvmmodel/RulesRefresher.java#L130-L131).

Would you be ok to refactor your PR accordingly?

Reply:
@kaikreuzer sure! I've just pushed an update which switches to that patttern. Hope that is better.

(BTW I originally added the dedicated tracker to the StartLevelService because I didn't want to replicate the functionality that it provided - e.g. storing all the ready markers and determining the singular current start level. I've just used a much simpler implementation however.)